src/Integration.Vsix/Analysis/SingleFileMonitor.cs

```diff
@@ -119,6 +119,26 @@ private static void EnsureDirectoryExists(string filePath, IFileSystem fileSyste
             }
         }
 
+        private bool IsDuplicateChangeEvent(string fullPath)
```

Comment:
From my understanding the only event that has duplicate reports is the change event so I seperated this in a different method.